appsec/src/extension/tags.c

```diff
@@ -490,7 +494,12 @@ static void _add_basic_tags_to_meta(
     zend_array *meta_ht = Z_ARRVAL_P(meta);
 
     _dd_http_client_ip(meta_ht);
-    _dd_request_headers(meta_ht, _server, &_relevant_basic_headers);
+
+    HashTable *headers = &_relevant_basic_headers;
+    if (_user_event_triggered) {
```

Comment:
Hiding this inside `_add_basic_tags_to_meta` really changes the meaning of it, maybe it would make more sense to refactor these functions to allow the caller, through one of the arguments, to define what has to be sent and then keep the logic of what is being sent contained to `dd_tags_add_tags`. It doesn't make a practical difference, but it's cleaner than unexpected behaviours determined by globals.